src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,57 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+type Props = {
+  searchInputValue: string;
+  setSearchInputValue: (value: string) => void;
+  selectedOption: string;
+  setSelectedOption: (option: string) => void;
+};
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+export const TodoFilter: React.FC<Props> = ({
+  searchInputValue,
+  setSearchInputValue,
+  selectedOption,
+  setSelectedOption,
+}) => {
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={selectedOption}
+            onChange={e => setSelectedOption(e.target.value)}
```

Comment:
The naming convention for event handler functions should follow the pattern `handle` + `EventName`. Consider renaming `setSelectedOption` to something like `handleOptionChange` to better reflect its purpose as an event handler.